### ISSUE TITLE
no-namespace: allow global augmentation

### ISSUE THREAD

**Issue by ajafff** [palantir/tslint#3157]
### Bug Report

- __TSLint version__: 5.6.0
- __TypeScript version__: 2.5.1
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
declare global { // should not produce an error
}
```

with `tslint.json` configuration:

```json
{
    "no-namespace": true
}
```

#### Actual behavior

`'namespace' and 'module' are disallowed`

#### Expected behavior
No error
